services/streammanager/kafka/kafkamanager.go

```diff
@@ -362,8 +363,10 @@ func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*Produ
 		return nil, err
 	}
 
-	// @TODO embedAvroSchemaID should come from control plane (i.e. destination config)
 	embedAvroSchemaID := config.GetBool("ROUTER_KAFKA_EMBED_AVRO_SCHEMA_ID_"+strings.ToUpper(destination.ID), false)
+	if !embedAvroSchemaID {
```

Comment:
The default value is false right? Wouldn't that break backwards compatibility?

Reply:
It's OK we have this inside a Kubernetes secret only. If you pass `true` in the Kubernetes secret that is going to be honoured and we won't read from control plane.

If you pass `false` or omit it then we try to read from control plane. If you get `true` from control plane it means that the "new way" of configuring this is live so it's OK.

The only ones that have this as `true` is Cars on their `dev` and they are waiting for this to be configurable from CP so it's OK :+1: